### PR TITLE
Custom registration bugs when using consent fields

### PR DESCRIPTION
**Pull request by bguyza**
Required for https://github.com/FusionAuth/fusionauth-app/pull/98

### CODE REVIEW

File: src/main/java/io/fusionauth/domain/form/Form.java [hunk 1]
```diff
@@ -46,6 +46,19 @@
...
+  public Form(Form other) {
+    this.data = other.data;
+    this.id = other.id;
+    this.insertInstant = other.insertInstant;
+    this.lastUpdateInstant = other.lastUpdateInstant;
+    this.name = other.name;
+    this.steps = other.steps;
+    this.type = other.type;
+  }
+
```
Comment (bguyza) [inline src/main/java/io/fusionauth/domain/form/Form.java:61]: I added a copy constructor so I could create an `updateForm` method in the test builder using the consumer copy pattern we use for other domain objects.

File: src/main/java/io/fusionauth/domain/form/Form.java [hunk 2]
```diff
@@ -46,6 +46,19 @@
...
+  public Form(Form other) {
+    this.data = other.data;
+    this.id = other.id;
+    this.insertInstant = other.insertInstant;
+    this.lastUpdateInstant = other.lastUpdateInstant;
+    this.name = other.name;
+    this.steps = other.steps;
+    this.type = other.type;
+  }
```
Comment (robotdan) [inline src/main/java/io/fusionauth/domain/form/Form.java:60]: Can you add the `@JacksonConstructor` annotation on the no-arg constructor?

Comment (robotdan) [inline src/main/java/io/fusionauth/domain/form/Form.java:60]: This copy constructor is invalid.

**Review by robotdan — DISMISSED** [review]

**Review by robotdan — COMMENTED** [review]
Added comments

**Review by robotdan — APPROVED** [review]

**Pull request merged**